src/Generator/SubProfileGenerator.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Drupal\lightning\Generator;
+
+use Drupal\Console\Core\Generator\Generator;
+
+class SubProfileGenerator extends Generator {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function generate(
+    $profile,
+    $machine_name,
+    $profile_path,
+    $description,
+    $dependencies,
+    $excluded_dependencies
+  ) {
+    $dir = $profile_path . '/' . $machine_name;
+
+    if (file_exists($dir)) {
+      if (!is_dir($dir)) {
+        throw new \RuntimeException(
+          sprintf(
```

Comment:
Can we call realpath() when we define $dir, rather than repeatedly calling it in sprintf()?